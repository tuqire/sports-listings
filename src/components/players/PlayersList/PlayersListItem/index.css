.players-list-item h6 {
  line-height: 1.5;
}
