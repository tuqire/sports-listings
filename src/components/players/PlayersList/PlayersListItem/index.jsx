import React from 'react'
import './index.css'

export default ({ player }) =>
  <div className="players-list-item">
    <h4>{player.firstname} {player.lastname}</h4>
    <h6 className="h6 text-muted small">{player.shortname}</h6>
    <img
      className="img-fluid mt-3" src={player.picture}
      alt={`picture of ${player.firstname} ${player.lastname}`}
    />
  </div>
