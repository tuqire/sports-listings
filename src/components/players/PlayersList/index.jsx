import React from 'react'
import PlayersListItem from './PlayersListItem'

export default props =>
  <div className="row">
    {props.players.map(player =>
      <div key={player.shortname} className="col-6 col-md-4 col-lg-3 mt-5 mb-3">
        <PlayersListItem player={player} />
      </div>
    )}
  </div>
