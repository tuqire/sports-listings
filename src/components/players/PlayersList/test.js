import React from 'react'
import Enzyme, { mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import PlayersList from './index'

Enzyme.configure({ adapter: new Adapter() })

const mockProps = {
  players: [{
      firstname: 'firstname 1',
      lastname: 'lastname 1',
      shortname: 'shortname 1',
      picture: 'picture 1'
    }, {
      firstname: 'firstname 2',
      lastname: 'lastname 2',
      shortname: 'shortname 2',
      picture: 'picture 2'
    }, {
      firstname: 'firstname 3',
      lastname: 'lastname 3',
      shortname: 'shortname 3',
      picture: 'picture 3'
    }
  ]
}

describe('components/players/PlayersList', () => {
  describe('#PlayersList', () => {
    describe('matches snapshots', () => {
      it('when nothing in store', () => {
        const component = mount(<PlayersList players={[]} />)
        expect(component).toMatchSnapshot()
      })

      it('when players in store', () => {
        const component = mount(<PlayersList {...mockProps} />)
        expect(component).toMatchSnapshot()
      })
    })
  })
})
