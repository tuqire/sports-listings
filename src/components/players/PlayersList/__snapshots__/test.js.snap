// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/players/PlayersList #PlayersList matches snapshots when nothing in store 1`] = `
<Component
  players={Array []}
>
  <div
    className="row"
  />
</Component>
`;

exports[`components/players/PlayersList #PlayersList matches snapshots when players in store 1`] = `
<Component
  players={
    Array [
      Object {
        "firstname": "firstname 1",
        "lastname": "lastname 1",
        "picture": "picture 1",
        "shortname": "shortname 1",
      },
      Object {
        "firstname": "firstname 2",
        "lastname": "lastname 2",
        "picture": "picture 2",
        "shortname": "shortname 2",
      },
      Object {
        "firstname": "firstname 3",
        "lastname": "lastname 3",
        "picture": "picture 3",
        "shortname": "shortname 3",
      },
    ]
  }
>
  <div
    className="row"
  >
    <div
      className="col-6 col-md-4 col-lg-3 mt-5 mb-3"
      key="shortname 1"
    >
      <Component
        player={
          Object {
            "firstname": "firstname 1",
            "lastname": "lastname 1",
            "picture": "picture 1",
            "shortname": "shortname 1",
          }
        }
      >
        <div
          className="players-list-item"
        >
          <h4>
            firstname 1
             
            lastname 1
          </h4>
          <h6
            className="h6 text-muted small"
          >
            shortname 1
          </h6>
          <img
            alt="picture of firstname 1 lastname 1"
            className="img-fluid mt-3"
            src="picture 1"
          />
        </div>
      </Component>
    </div>
    <div
      className="col-6 col-md-4 col-lg-3 mt-5 mb-3"
      key="shortname 2"
    >
      <Component
        player={
          Object {
            "firstname": "firstname 2",
            "lastname": "lastname 2",
            "picture": "picture 2",
            "shortname": "shortname 2",
          }
        }
      >
        <div
          className="players-list-item"
        >
          <h4>
            firstname 2
             
            lastname 2
          </h4>
          <h6
            className="h6 text-muted small"
          >
            shortname 2
          </h6>
          <img
            alt="picture of firstname 2 lastname 2"
            className="img-fluid mt-3"
            src="picture 2"
          />
        </div>
      </Component>
    </div>
    <div
      className="col-6 col-md-4 col-lg-3 mt-5 mb-3"
      key="shortname 3"
    >
      <Component
        player={
          Object {
            "firstname": "firstname 3",
            "lastname": "lastname 3",
            "picture": "picture 3",
            "shortname": "shortname 3",
          }
        }
      >
        <div
          className="players-list-item"
        >
          <h4>
            firstname 3
             
            lastname 3
          </h4>
          <h6
            className="h6 text-muted small"
          >
            shortname 3
          </h6>
          <img
            alt="picture of firstname 3 lastname 3"
            className="img-fluid mt-3"
            src="picture 3"
          />
        </div>
      </Component>
    </div>
  </div>
</Component>
`;
